### PR TITLE
fdk-aac-encoder 0.6.2 (new formula)

### PR DESCRIPTION
**Pull request by jrnewell**

### CODE REVIEW

File: Library/Formula/fdkaac.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+require 'formula'
```
Comment (bfontaine) [inline Library/Formula/fdkaac.rb]: You can remove this line.

File: Library/Formula/fdkaac.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+class Fdkaac < Formula
+  desc "Command-line encoder frontend for libfdk-aac"
+  homepage "https://github.com/nu774/fdkaac"
+  url "https://github.com/nu774/fdkaac/archive/v0.6.2.tar.gz"
+  version "0.6.2"
```
Comment (bfontaine) [inline Library/Formula/fdkaac.rb]: You can remove this, Homebrew detects it from the URL.

File: Library/Formula/fdkaac.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  sha256 "de758d6e651e81e9be89d2972612fc5b96cb70321234c3339f35483b636458ad"
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
+  depends_on 'pkg-config' => :build
```
Comment (bfontaine) [inline Library/Formula/fdkaac.rb]: Please use double-quoted strings everywhere.

File: Library/Formula/fdkaac.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  depends_on 'fdk-aac'
+
+  def install
+    system "autoreconf -i"
```
Comment (bfontaine) [inline Library/Formula/fdkaac.rb]: Use `system "autoreconf", "-i"` instead of `system "autoreconf -i"`.

File: Library/Formula/fdkaac.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+    system "./configure", "--disable-debug",
+                          "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}",
+                          "--mandir=#{man}"
+    system "make", "install"
+  end
```
Comment (bfontaine) [inline Library/Formula/fdkaac.rb]: Could you please add a test with a `test do` block? Thanks!

**Comment by DomT4** [comment]
Test is breaking:

```
Error: fdkaac: failed
Failed executing: /usr/local/Cellar/fdkaac/0.6.2/bin/fdkaac --help
```

Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

File: Library/Formula/fdkaac.rb [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    frequency = 440.0
+    max_amplitude = 0.2
+
+    position_in_period = 0.0
+    position_in_period_delta = frequency / sample_rate
+
+    samples = [].fill(0.0, 0, num_samples)
+
+    num_samples.times do |i|
+      samples[i] = Math::sin(position_in_period * two_pi) * max_amplitude
```
Comment (bfontaine) [inline Library/Formula/fdkaac.rb]: Use a dot here instead of `::`.

File: Library/Formula/fdkaac.rb [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+      position_in_period += position_in_period_delta
+
+      if(position_in_period >= 1.0)
```
Comment (bfontaine) [inline Library/Formula/fdkaac.rb]: You can drop the parentheses here.

File: Library/Formula/fdkaac.rb [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+        position_in_period -= 1.0
+      end
+    end
+
+    samples.map! do |sample|
+      (sample * 32767.0).round
+    end
+
+    File.open("#{testpath}/tone.pcm", "wb") {
```
Comment (bfontaine) [inline Library/Formula/fdkaac.rb]: Use `do`/`end` for multi-line blocks

**Comment by DomT4** [comment]
I'm not sure here that it's not deeply confusing to have a `fdkaac` formula that depends on `fdk-aac`. It'd be very easy to mistype that and end up with an unexpected package.

**Comment by mistydemeo** [comment]
I agree, I think the name is really confusing. I came to this issue thinking, "don't we already have a formula for this?"

**Comment by jrnewell** [comment]
I had the same hesitation about the naming as well, but I was trying to follow formula cookbook which says "Name the formula like the project markets the product.".  IMO, a better name for this would be something like ``fdk-aac-encoder``

As to relevancy, there are formulae for ``vorbis-tools`` and ``opus-tools`` which basically provide the same functionality as this - a CLI encoder (plus a few other tools) wrapping ``libvoribis`` and ``opus`` respectively.

**Comment by DomT4** [comment]
We bend the usual naming if two formulae sound *too* similar (The two `aescrypt` formula for example are called the same thing upstream but we named one differently).

> fdk-aac-encoder

This would be fine with me at least.

File: Library/Formula/fdk-aac-encoder.rb [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+    system "./configure", "--disable-debug",
+                          "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}",
+                          "--mandir=#{man}"
+    system "make", "install"
+  end
+
+  def generate_tone_pcm
+    ohai "generating test audio file"
```
Comment (DomT4) [inline Library/Formula/fdk-aac-encoder.rb]: Let's not use `ohai` here. It's not likely to take long enough to merit needing to bug the user about.

File: Library/Formula/fdk-aac-encoder.rb [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+
+    system "./configure", "--disable-debug",
+                          "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}",
+                          "--mandir=#{man}"
+    system "make", "install"
+  end
+
+  def generate_tone_pcm
```
Comment (DomT4) [inline Library/Formula/fdk-aac-encoder.rb]: Let's inline this into the test rather than using a `def` block, Thanks!

**Comment by DomT4** [comment]
@BrewTestBot test this please

**Comment by DomT4** [comment]
@BrewTestBot test this please

**Comment by DomT4** [comment]
Nobody raised an objection here, so merged in https://github.com/Homebrew/homebrew/commit/9aa419dfa664a815097d32cde168a68b30cd7629. Thank you for your contribution to Homebrew, we appreciate it! :smiley_cat:.

Just a nit, but new formulae should have their commit messages styled `fdk-aac-encoder 0.6.2 (new formula)` or similar, Thanks!

**Pull request closed without merging**